File: src-tauri/src/models/inventory.rs
use serde::{Deserialize, Serialize};
use sqlx::prelude::FromRow;

#[derive(Debug, Deserialize, FromRow, Serialize, Clone)]
pub struct Inventory {
    pub id: String,
    pub amount: i64,
    pub item_id: String,
}

#[derive(Debug, Deserialize, FromRow, Serialize, Clone)]
pub struct InventoryWithItem {
    pub id: Option<String>,
    pub item_name: String,
    pub item_picture: Option<String>,
    pub item_id: String,
    pub amount: Option<i64>,
    pub game_id: String
}


#[derive(Debug, Deserialize, FromRow, Serialize, Clone)]
pub struct InventoryDto {
    pub id: Option<String>,
    pub amount: i64,
    pub item_id: String,
}

File: src/models/inventory.ts
interface InventoryWithItem {
    id?: string;
    item_name: string;
    item_picture?: string;
    item_picture_b64: string;
    item_id: string;
    amount?: number;
    game_id: string;
}

interface InventoryDto {
    id?: string;
    amount: number;
    item_id: string;
}

interface Inventory {
    id: string;
    amount: number;
    item_id: string;
}